lib/Appinfo/Application.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * @copyright Copyright (c) 2020 Vinzenz Rosenkranz <vinzenz.rosenkranz@posteo.de>
+ *
+ * @author Vinzenz Rosenkranz <vinzenz.rosenkranz@posteo.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Files3d\AppInfo;
+
+use OCA\Viewer\Event\LoadViewer;
+use OCA\Viewer\Listener\LoadViewerScript;
```

Comment:
This will just load the viewer (which already listen to this event and load itself), what you want is to load your files_3d javascript, so you can import a clone of this `LoadViewerScript` file, and rename it appropriately to you rapp and use it to load your own Scripts :)

```js
Util::addScript(Application::APP_ID, 'yourscriptname');
```

Reply:
I replaced the `addEventListener` in Application.php with my own class
```php
$eventDispatcher->addServiceListener(LoadViewer::class, LoadFiles3dScript::class);
```

and created a `LoadFiles3dScript` class which is more or less a copy of the `LoadViewerClass` file

```php
namespace OCA\Files3d\Listener;

use OCA\Files3d\AppInfo\Application;
use OCA\Files\Event\LoadAdditionalScriptsEvent;
use OCA\Viewer\Event\LoadViewer;
use OCP\EventDispatcher\Event;
use OCP\EventDispatcher\IEventListener;
use OCP\Util;

class LoadFiles3dScript implements IEventListener {
    public function handle(Event $event): void {
        if (!($event instanceof LoadViewer || $event instanceof LoadAdditionalScriptsEvent)) {
            return;
        }

        Util::addScript(Application::APP_ID, 'files3d');
    }
}
```
DId I miss something?

Sorry for my dumb questions :sweat_smile:

Reply:
Looks good as far as I overlooked :)

Reply:
> ```php
> if (!($event instanceof LoadViewer || $event instanceof LoadAdditionalScriptsEvent)) {
> ```

I guess this should be only this as you're only listening to the LoadViewer event :)
```php
if (!($event instanceof LoadViewer)) {
```

Reply:
That's great. I pushed the latest changes. Thanks for your time and help :rocket: